### ISSUE TITLE
Traceback when publishing with a skipped package

### ISSUE THREAD

**Issue by soininen** [spine-tools/spine-conductor#34]
I created new releases while skipping `spine_items`:

```
conduct release --exclude spine_items --bump patch -c release.toml
```

When trying to publish with

```
conduct publish --pkgtags pkgtags.json -c release.toml
```

encountered this Traceback:

```
┌─────────────────────────────── Traceback (most recent call last) ────────────────────────────────┐
│ spine-conductor\orchestra\__main__.py:78 in publish                   │
│                                                                                                  │
│    75 ):                                                                                         │
│    76 │   """Push Git tags to GitHub and publish packages to PyPI"""                             │
│    77 │   conf = read_conf(f"{config}")                                                          │
│ >  78 │   publish_tags_whls(conf, pkgtags)                                                       │
│    79                                                                                            │
│    80                                                                                            │
│    81 _xtest_doc_ref = (                                                                         │
│                                                                                                  │
│ ┌─────────────────────────────────────────── locals ───────────────────────────────────────────┐ │
│ │    conf = {                                                                                  │ │
│ │           │   'packagename_regex': 'spine(toolbox|(db){0,1}[_-][a-z]+)',                     │ │
│ │           │   'pkgname_re': re.compile('spine(toolbox|(db){0,1}[_-][a-z]+)'),                │ │
│ │           │   'repos': {'spinetoolbox': '.', 'spine_items': '../spine-items',                │ │
│ │           'spine_engine': '../spine-engine', 'spinedb_api': '../Spine-Database-API'},        │ │
│ │           │   'dependency_graph': {'spinetoolbox': ['spine_items', 'spine_engine',           │ │
│ │           'spinedb_api'], 'spine_items': ['spinetoolbox', 'spine_engine', 'spinedb_api'],    │ │
│ │           'spine_engine': ['spinedb_api'], 'spinedb_api': []},                               │ │
│ │           │   'default_branch': 'master',                                                    │ │
│ │           │   'branches': {'spinetoolbox': 'release', 'spine_items': 'release',              │ │
│ │           'spine_engine': 'release', 'spinedb_api': 'release'},                              │ │
│ │           │   'workflow': {'repo': 'spine-tools/spine-conductor', 'file':                    │ │
│ │           'test-n-publish.yml'}                                                              │ │
│ │           }                                                                                  │ │
│ │  config = WindowsPath('release.toml')                                                        │ │
│ │     ctx = <click.core.Context object at 0x0000026A2B245E80>                                  │ │
│ │ pkgtags = WindowsPath('pkgtags.json')                                                        │ │
│ └──────────────────────────────────────────────────────────────────────────────────────────────┘ │
│                                                                                                  │
│ spine-conductor\orchestra\publish.py:111 in publish_tags_whls         │
│                                                                                                  │
│   108 │   tags = json.loads(pkgtags.read_text())                                                 │
│   109 │   for pkg, repo_path in CONF["repos"].items():                                           │
│   110 │   │   repo = Repo(repo_path)                                                             │
│ > 111 │   │   push_tags(CONF, pkg, repo, tags[remote_name(repo)])                                │
│   112 │   dispatch_workflow(CONF, pkgtags)                                                       │
│   113                                                                                            │
│                                                                                                  │
│ ┌─────────────────────────────────────────── locals ───────────────────────────────────────────┐ │
│ │      CONF = {                                                                                │ │
│ │             │   'packagename_regex': 'spine(toolbox|(db){0,1}[_-][a-z]+)',                   │ │
│ │             │   'pkgname_re': re.compile('spine(toolbox|(db){0,1}[_-][a-z]+)'),              │ │
│ │             │   'repos': {'spinetoolbox': '.', 'spine_items': '../spine-items',              │ │
│ │             'spine_engine': '../spine-engine', 'spinedb_api': '../Spine-Database-API'},      │ │
│ │             │   'dependency_graph': {'spinetoolbox': ['spine_items', 'spine_engine',         │ │
│ │             'spinedb_api'], 'spine_items': ['spinetoolbox', 'spine_engine', 'spinedb_api'],  │ │
│ │             'spine_engine': ['spinedb_api'], 'spinedb_api': []},                             │ │
│ │             │   'default_branch': 'master',                                                  │ │
│ │             │   'branches': {'spinetoolbox': 'release', 'spine_items': 'release',            │ │
│ │             'spine_engine': 'release', 'spinedb_api': 'release'},                            │ │
│ │             │   'workflow': {'repo': 'spine-tools/spine-conductor', 'file':                  │ │
│ │             'test-n-publish.yml'}                                                            │ │
│ │             }                                                                                │ │
│ │       pkg = 'spine_items'                                                                    │ │
│ │   pkgtags = WindowsPath('pkgtags.json')                                                      │ │
│ │      repo = <git.repo.base.Repo 'spine-items\\.git'>          │ │
│ │ repo_path = '../spine-items'                                                                 │ │
│ │      tags = {                                                                                │ │
│ │             │   'Spine-Toolbox': '0.8.4',                                                    │ │
│ │             │   'spine-engine': '0.24.3',                                                    │ │
│ │             │   'Spine-Database-API': '0.31.5'                                               │ │
│ │             }                                                                                │ │
│ └──────────────────────────────────────────────────────────────────────────────────────────────┘ │
└──────────────────────────────────────────────────────────────────────────────────────────────────┘
KeyError: 'spine-items'

```

**Comment by suvayu** [comment]
I think the current behaviour is that you have to repeat the `--skip <pkg>` argument also for publish.  But you are right in not expecting that, since `pkgtags.json` clearly specifies which packages to release.

**Comment by soininen** [comment]
`conduct publish --help` does not mention the `--exclude` flag so I though it would pick the packages automatically from `pkgtags.json` in any case.

**Comment by suvayu** [comment]
Oh! That's a double bug then!

**Comment by soininen** [comment]
Does it even make sense to have `--exclude` in `publish`? What happens if you exclude packages that you did not exclude when running `release`?

**Comment by suvayu** [comment]
fixed in af6f872
